subprojects/workers/src/main/java/org/gradle/workers/WorkerExecutor.java

```diff
@@ -20,26 +20,38 @@
 import org.gradle.api.Incubating;
 
 /**
- * Worker Executor.
+ * Allows work to be submitted for asynchronous execution.  Work should be submitted with a {@link Runnable} class
```

Comment:
We should explain some of the concepts mentioned in the reference manual here as well. I.e. why would I use this API?